### PR TITLE
Updates to nds2-client-0.16.6 updates to nds2-client-swig-0.16.8

### PR DESCRIPTION
**Pull request by emaros**
#### Description

This updates the Portfiles for the most recent public releases of nds2-client and nds2-client-swig.
Use of Java PortGroup has been added.

This should supersede pull request #8022

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.12
Xcode 9.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by cjones051073** [comment]
Please update your commit message to follow our guidelines

Portname: description

**Comment by cjones051073** [comment]
Could you please you two independent commits for each of the port updates

**Comment by emaros** [comment]
> Could you please you two independent commits for each of the port updates

I have made the attempt, but am now getting a failure (see above). Would you please help me understand what I need to do to address the newly created problem?

File: science/nds2-client-swig/Portfile [hunk 1]
```diff
@@ -92,8 +106,13 @@ subport ${basename}-matlab {
...
                             bindings, modules, and scripts.
 
-  # Need GNU Classpath to get jni.h header
-  depends_build-append      port:swig-java port:gnu-classpath
+  if {[variant_isset swig3]} {
+    depends_build-append      port:swig3-java
+  }
+  java.version              1.7+
+  # Set fallback to an LTS Java version
+  java.fallback             openjdk8
```
Comment (chrstphrchvz) [inline science/nds2-client-swig/Portfile:114]: The newest build errors are due to using `java.version`/`java.fallback` without including the Java portgroup for the MATLAB subport.

In #8022 I thought the Java build dependency might only be needed for the Java subport rather than also the MATLAB/Octave subports, but I am not sure.

**Review by chrstphrchvz — CHANGES_REQUESTED** [review]

File: science/nds2-client-swig/Portfile (hunk 1 above)
Comment (emaros) [reply science/nds2-client-swig/Portfile:109]: In my latest commit, I have added the PortGroup java to the matlab subport as finding JAVA is currently needed for the installation of the matlab .m files. This causes a lint failure which I do not understand how to correct.

Comment (chrstphrchvz) [reply science/nds2-client-swig/Portfile:109]: The `repeats inclusion of PortGroup java` linting error is expected/can be ignored; it's a (known?) limitation of `port lint` when checking subports.

File: science/nds2-client-swig/Portfile [hunk 2]
```diff
@@ -4,31 +4,28 @@ PortSystem        1.0
...
 
 name              nds2-client-swig
-version           0.16.3
+version           0.16.8
 revision          0
 categories        science
 platforms         darwin
 license           GPL-2
-maintainers       {ligo.org:ed.maros @emaros} openmaintainer
+maintainers       ligo.org:ed.maros openmaintainer
```
Comment (cjones051073) [inline science/nds2-client-swig/Portfile:12]: Why remove your github handle ?

Comment (emaros) [reply science/nds2-client-swig/Portfile]: Reverted change

File: science/nds2-client-swig/Portfile [hunk 3]
```diff
@@ -141,7 +161,7 @@ foreach v {27 35 36 37 38} {
   set python.branch           [string range ${python.version} 0 end-1].[string index ${python.version} end]
   set python.bin              ${prefix}/bin/python${python.branch}
   set python.prefix           ${frameworks_dir}/Python.framework/Versions/${python.branch}
-  set python.site_packages    "${python.prefix}/lib/python${python.branch}/site-packages"
+    set python.site_packages    "${python.prefix}/lib/python${python.branch}/site-packages"
```
Comment (ryandesign) [inline science/nds2-client-swig/Portfile:164]: You should revert this change. Although I would like to see the whitespace of this port fixed to conform to the modeline (i.e. 4 spaces per indentation) it should be done to the entire portfile at once without any functional changes, so let's leave that for another day.
```suggestion
  set python.site_packages    "${python.prefix}/lib/python${python.branch}/site-packages"
```

File: science/nds2-client/Portfile [hunk 4]
```diff
@@ -16,24 +16,25 @@ long_description \
...
-checksums         rmd160  9653ad19cc2af2d194c8331e5cac99c6beb95119 \
-                  sha256  485ce5cc1650f384897e5ddcd259ad2f4f6b63e309372990b997d86b54cf529e \
-                  size    535845
+checksums         rmd160  d838d9602fc7e5258fae67b47fddaca99b58ea9e \
+                  sha256  a6d83cc0f96f63ca7e045f3923dc178ad48a4869d61d5e2086d184c7965e5e15 \
+                  size    583936
 
-depends_build-append \
+depends_build \
+                  port:cmake \
```
Comment (ryandesign) [inline science/nds2-client/Portfile:28]: You should revert this change. It was correct to append to the dependencies, so as not to overwrite any dependencies that the portgroups you include might set -- such as the cmake dependency that the cmake portgroup adds for you.
```suggestion
depends_build-append \
```

File: science/nds2-client/Portfile [hunk 5]
```diff
@@ -16,24 +16,25 @@ long_description \
...
                   port:doxygen \
-                  path:bin/dot:graphviz \
-                  port:pkgconfig
+                  port:graphviz \
```
Comment (ryandesign) [inline science/nds2-client/Portfile:30]: You should revert this change. It was correct to depend on graphviz using the `path` syntax because there are multiple ports that provide the graphviz software -- graphviz and graphviz-devel -- and this syntax allows either of them to satisfy the dependency.
```suggestion
                  path:bin/dot:graphviz \
```

File: science/nds2-client/Portfile [hunk 6]
```diff
@@ -16,24 +16,25 @@ long_description \
...
+                  port:pkgconfig \
+                  port:boost
 
 default_variants  +gssapi
 
 cmake.out_of_source yes
 
-compiler.cxx_standard 2011
```
Comment (ryandesign) [inline science/nds2-client/Portfile:35]: Are you saying that this software no longer uses the C++ language? If it does still use C++, then you should still indicate what version of the language it requires.

File: science/nds2-client-swig/Portfile [hunk 7]
```diff
@@ -48,6 +45,15 @@ configure.args    -DPYTHON=false \
                   -DCMAKE_BUILD_TYPE=RelWithDebInfo \
                   -DCMAKE_EXPORT_COMPILE_COMMANDS=1
 
+variant swig3 description {Use SWIG 3.x for generation of extensions} {
+    
+    depends_build-append    port:swig3
+    configure.args-append \
+        -DSWIG_EXECUTABLE:FILEPATH=${prefix}/bin/swig3
+}
```
Comment (ryandesign) [inline science/nds2-client-swig/Portfile:53]: Why have you added a swig3 variant? What happens if the variant is deselected? I see several places in the port that reference swig even if the variant is not selected.

File: science/nds2-client-swig/Portfile (hunk 3 above)
Comment (emaros) [reply science/nds2-client-swig/Portfile]: I have added this to my latest commit

File: science/nds2-client/Portfile (hunk 4 above)
Comment (emaros) [reply science/nds2-client/Portfile]: I have added this to my latest commit

File: science/nds2-client/Portfile (hunk 5 above)
Comment (emaros) [reply science/nds2-client/Portfile]: I have added this to my latest commit

File: science/nds2-client/Portfile (hunk 6 above)
Comment (emaros) [reply science/nds2-client/Portfile]: I have added this back in my latest commit

File: science/nds2-client-swig/Portfile (hunk 7 above)
Comment (emaros) [reply science/nds2-client-swig/Portfile:53]: I was setting up for the future for having a swig4 variant. Currently, swig(swig 4.x) does not work but steps are being taken to support it.

Comment (ryandesign) [reply science/nds2-client-swig/Portfile:53]: If I try to install the port and disable the swig3 variant, it fails:
```
CMake Error at /opt/local/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find SWIG (missing: SWIG_EXECUTABLE SWIG_DIR) (Required is at
  least version "3.0.7")
```
So I would not make this a variant.

Even if it were compatible with swig 4, I don't see why the user would want to specify which swig to use. Just make the port depend on the most recent swig with which it is compatible.

Comment (emaros) [reply science/nds2-client-swig/Portfile]: Variant has been eliminated

**Comment by emaros** [comment]
Are there any additional changes that are required for these Portfiles to be accepted?

**Comment by chrstphrchvz** [comment]
Pinging @emaros—there is a conflict to resolve in the portfile for nds2-client-swig

**Comment by reneeotten** [comment]
I have made the requested changes, update to latest versions and a few other modifications ; let's see what the CI thinks about this and get this merged...

**Comment by reneeotten** [comment]
this does timeout on the CI systems - but whatever started builds correctly... I'll merge this and try to fix if something comes up
